src/app/components/pictureServerComponents/uploadButton.tsx

```
@@ -0,0 +1,41 @@
+"use client";
+
+import { useState } from "react";
+
+export default function ImageUploaderButton({ uploadImage, setUploadedFileName }) {
+  const [selectedFile, setSelectedFile] = useState(null);
+
+  const handleFileChange = (event) => {
+    setSelectedFile(event.target.files[0]);
+  };
+
+  const handleUpload = async () => {
+    if (!selectedFile) {
+      alert("Please select a file first.");
+      return;
+    }
+
+    const formData = new FormData();
+    formData.append("image", selectedFile);
+
+    const response = await uploadImage(formData);
+    if (response.success) {
+      setUploadedFileName(response.data.filename); // Update parent state
+      alert(`Image uploaded successfully! Filename: ${response.data.filename}`);
```

This will be used by the users, right? Then there probably shouldn't be an alert each time it is used. One should probably instead throw an exception at the form.

---

i agree, i was thinking of making a more semless feedback system before realese